docs/tracing.md

```diff
@@ -0,0 +1,300 @@
+| | Forwarder |    VSOCK   |     |Agent|  | |
+| +-----------+    Channel |     +-----+  | |
+|                          +--------------+ |
++-------------------------------------------+
+```
+
+# Agent tracing terminology
+
+## Trace modes
+
+The agent supports two different tracing modes:
+
+| Trace mode | Description | Use-case | Setting | VM shutdown controller | Notes |
+|-|-|-|-|-|-|
+| Static | Traces from agent start to agent shutdown. | Holistic view of agent activities. Possibly more suited to short-running containers. | `trace_mode = "static"` | agent | Default trace mode. |
+| Dynamic | Traces begin when the ttRPC `StartTracing()` API is called and end when the corresponding `StopTracing()` API is called. | On-demand (partial) tracing. More appropriate for long-running containers. | `trace_mode = "dynamic"` | runtime | Only "isolated" type supported. |
+
+## Trace types
+
+Each trace mode is sub-divided into two different types:
+
+| Trace type | Description | Use-case | Setting | Notes |
+|-|-|-|-|-|
+| isolated | The traces only apply to the agent: the first span is generated at agent startup and the last at agent shutdown. | Observing agent lifespan. | `trace_type = "isolated"` | Default trace type. |
+| collated | In this mode, spans are associated with their Kata runtime initiated counterparts. | Understanding how the runtime and agent interact. | `trace_type = "collated"` | Also requires runtime tracing to be enabled in `configuration.toml` (`enable_tracing=true`). |
+
+# Prerequisites
+
+- You must have a trace collector running on the host.
+
+  The quickest and simplest method for doing this is to use the
+  [Jaeger "all-in-one" Docker image][jaeger-all-in-one].
+
+- If you wish to trace the agent, you must start the
+  [trace forwarder][trace-forwarder].
+
+> **Notes:**
+>
+> - If agent tracing is enabled but the forwarder is not running,
+>   the agent will exit (panic) with an error message. Since the agent is not
+>   directly visible by the user, it is recommended to enable debug to capture
+>   this message in the logs.
+>
+> - The forwarder requires a trace collector (such as Jaeger) to be running
+>   before it is started. If a collector is not running, the trace forwarder
+>   will exit with an error.
+
+# Enable tracing
+
+By default, tracing is disabled for all components. To enable _any_ form of
+tracing the `enable_tracing` option must be enabled.
+
+> **Note:** 
+>
+> Enabling this option will only allow tracing for subsequently
+> started components.
+
+## Enable runtime tracing
+
+To trace the runtime, simply set the tracing option as shown:
+
+```toml
+[runtime]
+enable_tracing = true
+```
+
+## Enable agent tracing
+
+The agent may require more configuration than the runtime. However, the simplest way to trace the agent is to enable it tracing option:
+
+```toml
+[agent.kata]
+enable_tracing = true
+```
+
+If no other trace options are specified this will enable _static isolated_
+tracing.
+
```

Comment:
I'm only able to get agent tracing output with this exact configuration right now (described in the doc below):
```
[agent.kata]
enable_tracing = true
trace_mode = "static"
```
 If we want to merge this I think we should explicitly state to specify `trace_mode = "static"` for now, otherwise leave as pending/DNM.

Reply:
@cmaf I believe that @jodh-intel has decided to write the doc "as if it was all finished". I agree there are places where documenting the current limits would definitely be helpful.